app/src/main/java/org/mozilla/fenix/home/pocket/PocketCategoriesViewHolder.kt

```diff
@@ -55,12 +58,40 @@ class PocketCategoriesViewHolder(
         val categoriesSelections = components.appStore
             .observeAsComposableState { state -> state.pocketStoriesCategoriesSelections }.value
 
+        val wallpaperState = components.appStore
+            .observeAsComposableState { state -> state.wallpaperState }.value ?: WallpaperState.default
+
+        var selectedBackgroundColor: Color? = null
+        var unselectedBackgroundColor: Color? = null
+        var selectedTextColor: Color? = null
+        var unselectedTextColor: Color? = null
+        if (wallpaperState.areWallpaperCardColorsAvailable()) {
+            val cardColorLight = Color(wallpaperState.currentWallpaper.cardColorLight!!)
+            val cardColorDark = Color(wallpaperState.currentWallpaper.cardColorDark!!)
```

Comment:
Should this use `wallpaperCardColor` getter? Mostly just looking at `!!` and thinking why kotlin isn't smarter considering `areWallpaperCardColorsAvailable` checks for the nullability.

Reply:
I can't in this particular situation because both colors are actually needed to have:
- Light for selected and dark for unselected in light theme
- Dark for selected and light for unselected in dark theme

I, too, was disappointed that Kotlin wasn't able to detect that these were not null after the `areWallpaperCardColorsAvailable` check.

Reply:
I'm wondering if it's only because `wallpaperState` isn't kept in scope. You might be able to do something like:

```
with(wallpaperState) {
    if (areWallpaperCardColorsAvailable()) {
        cardColorLight = Color(currentWallpaper.cardColorLight)
...
    }
}
```

Reply:
no dice, unfortunately :/

Reply:
I also tried moving the `areWallpaperCardColorsAvailable` check to the `Wallpaper` class to no avail. 

It'd be cool if there was a way to do a combo `?.let` like:

(cardColorDark, cardColorLight)?.let {}

I think I have two potential solutions:

1. A solution inspired by Matt where we essentially wrap the colors in a `Pair` or a data class and do a null/let off of that returned object:

```
val wallpaperCardColors: Pair<Color, Color>?
    @Composable get() =
        Result.runCatching {
           Color(currentWallpaper.cardColorLight!!) to Color(currentWallpaper.cardColorDark!!)
        }.getOrNull()
```

2. Have a function on `WallpaperState` that runs a lambda if both of the card colors are not null:
```
@Composable
fun runIfWallpaperCardColorsAreAvailable(run: @Composable (cardColorLight: Color, cardColorDark: Color) -> Unit) {
    if (areWallpaperCardColorsAvailable()) {
        run(Color(currentWallpaper.cardColorLight!!), Color(currentWallpaper.cardColorDark!!
    }
}

wallpaperState.runIfWallpaperCardColorsAreAvailable { cardColorLight, cardColorDark ->
            ...
}
```

Reply:
I might be a bit biased towards the first, but I think the second is good too. There is some precedent in doing something similar like [runIfFragmentIsAttached](https://searchfox.org/mozilla-mobile/rev/fa471b2f8de50d396345e0a7f99f785d5acfe278/fenix/app/src/main/java/org/mozilla/fenix/ext/Fragment.kt#47) or [runIfReadyOrQueue](https://searchfox.org/mozilla-mobile/rev/fa471b2f8de50d396345e0a7f99f785d5acfe278/android-components/components/support/utils/src/main/java/mozilla/components/support/utils/RunWhenReadyQueue.kt#36)

Reply:
updated with the second approach